core/src/nodes/tasks/manager.rs

```diff
@@ -176,8 +174,14 @@ impl<I, O, H, E, HE, T, C> Manager<I, O, H, E, HE, T, C> {
         self.tasks.insert(task_id, TaskInfo { sender: tx, user_data });
 
         let task = Box::pin(Task::new(task_id, self.events_tx.clone(), rx, future, handler));
-        if let Some(threads_pool) = &mut self.threads_pool {
-            threads_pool.spawn_ok(task);
+        if let Some(executor) = &self.executor {
+            if executor.status().is_ok() {
+                if executor.spawn(task).is_err() {
+                    log::error!("Failed to spawn task for peer");
+                }
+            } else {
+                self.local_spawns.push(task);
+            }
```

Comment:
Just curious: Why not
```rust
if let Some(executor) = &self.executor {
    if let Err(e) = executor.spawn(task) {
        log::warn!("Failed to spawn task onto executor ({:?}). Resorting to in-place polling.", e);
        self.local_spawns.push(task);
    }           
} else {
    self.local_spawns.push(task);
}
```
?

Reply:
Calling `executor.spawn` entirely consumes the task.

Reply:
Oh, of course :man_facepalming: